Code/Framework/AzCore/AzCore/DOM/DomValue.cpp

```diff
@@ -0,0 +1,1241 @@
+                {
+                    result = visitor.Null();
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, int64_t>)
+                {
+                    result = visitor.Int64(arg);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, uint64_t>)
+                {
+                    result = visitor.Uint64(arg);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, double>)
+                {
+                    result = visitor.Double(arg);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, bool>)
+                {
+                    result = visitor.Bool(arg);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, AZStd::string_view>)
+                {
+                    result = visitor.String(arg, copyStrings ? Lifetime::Temporary : Lifetime::Persistent);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, AZStd::shared_ptr<const AZStd::string>>)
+                {
+                    result = visitor.RefCountedString(arg, copyStrings ? Lifetime::Temporary : Lifetime::Persistent);
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, ObjectPtr>)
+                {
+                    result = visitor.StartObject();
+                    if (result.IsSuccess())
+                    {
+                        const Object::ContainerType& object = GetObjectInternal();
+                        for (const Object::EntryType& entry : object)
+                        {
+                            result = visitor.Key(entry.first);
+                            if (!result.IsSuccess())
+                            {
+                                return;
+                            }
+                            result = entry.second.Accept(visitor, copyStrings);
+                            if (!result.IsSuccess())
+                            {
+                                return;
+                            }
+                        }
+                        result = visitor.EndObject(object.size());
+                    }
+                }
+                else if constexpr (AZStd::is_same_v<Alternative, ArrayPtr>)
+                {
+                    result = visitor.StartArray();
+                    if (result.IsSuccess())
+                    {
+                        const Array::ContainerType& arrayContainer = GetArrayInternal();
+                        for (const Value& entry : arrayContainer)
+                        {
+                            result = entry.Accept(visitor, copyStrings);
+                            if (!result.IsSuccess())
+                            {
+                                return;
+                            }
+                        }
+                        result = visitor.EndArray(arrayContainer.size());
```

Comment:
repeat: This should return if the array entry couldn't be visited, but instead break out of the for loop and then invoke EndArray, while aggregating the results of both the failed entry and the call to EndArray